sdks/python/apache_beam/io/external/xlang_jdbcio_it_test.py

```diff
@@ -0,0 +1,165 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# pytype: skip-file
+
+from __future__ import absolute_import
+
+import logging
+import typing
+import unittest
+
+from past.builtins import unicode
+
+import apache_beam as beam
+from apache_beam import coders
+from apache_beam.io.external.jdbc import WriteToJdbc
+from apache_beam.options.pipeline_options import PipelineOptions
+from apache_beam.testing.test_pipeline import TestPipeline
+
+# pylint: disable=wrong-import-order, wrong-import-position
+try:
+  import psycopg2
+except ImportError:
+  unittest.skip("no psycopg2 installed")
+  # pylint: enable=wrong-import-order, wrong-import-position
+
+JdbcTestRow = typing.NamedTuple(
+    "JdbcTestRow",
+    [
+        ("f_id", int),
+        ("f_real", float),
+        ("f_string", unicode),
+    ],
+)
+
+coders.registry.register_coder(JdbcTestRow, coders.RowCoder)
+
+ROW_COUNT = 10
+
+
+class JdbcExternalTransformTest(unittest.TestCase):
+  """Tests that exercise the cross-language JdbcIO Transform (implemented in java).
+
+  To run with the local expansion service and flink job server you need to build it,
+  e.g. via command:
+  ./gradlew :sdks:java:io:expansion-service:shadowJar
+  ./gradlew :runners:flink:1.10:job-server:shadowJar
+  and have flink1.10 master cluster running
+
+  If on development branch, it may be necessary to build java sdk docker image and tag
+  it with the latest released version.
+  ./gradlew :sdks:java:container:docker
+  docker tag apache/beam_java_sdk:x.xx.x.dev apache/beam_java_sdk:y.yy.y.dev
+
+  python setup.py nosetests \
+  --tests=apache_beam.io.external.jdbc_test:JdbcExternalTransformTest.test_read_and_write \
+  --test-pipeline-options="
+    --driver_class_name=org.postgresql.Driver
+    --jdbc_url=jdbc:postgresql://localhost:5432/postgres
+    --username=postgres
+    --password=postgres
+    --host=localhost
+    --port=5432
+    --database=postgres"
+  """
+  def __init__(self, *args, **kwargs):
+    super(JdbcExternalTransformTest, self).__init__(*args, **kwargs)
+
+    pipeline = TestPipeline()
+    #  Options for the external jdbc write transform
+    self.driver_class_name = pipeline.get_option('driver_class_name')
+    self.jdbc_url = pipeline.get_option('jdbc_url')
+    self.username = pipeline.get_option('username')
+    self.password = pipeline.get_option('password')
+
+    #  Required for table initialization and drop
+    self.host = pipeline.get_option('host')
+    self.port = pipeline.get_option('port')
+    self.database = pipeline.get_option('database')
+
+    #  The actual pipeline options
+    self.pipeline_options = {
+        'runner': 'FlinkRunner',
+        'flink_version': '1.10',
+        'flink_master': 'localhost:8081',
```

Comment:
If you specify FlinkRunner without specifying a flink_master a Flink instance should be started for you.

Reply:
Cool! I missed that info when reading portability docs.